src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/CollectionsMarshal.cs

```diff
@@ -16,5 +19,36 @@ public static class CollectionsMarshal
         /// </summary>
         public static Span<T> AsSpan<T>(List<T>? list)
             => list is null ? default : new Span<T>(list._items, 0, list._size);
+
+        public static ref TValue GetValueRef<TKey, TValue>(Dictionary<TKey, TValue> dictionary, [NotNull] TKey key) where TKey : notnull
```

Comment:
I agree.  I think this whole feature should really just be about exposing the FindValue functionality under the new location/name.  I believe @tannergooding raised objections to that in the API review mtg, so he should comment.

Reply:
I think it's fine to just have the `GetValueRef` that doesn't throw and that returns `Unsafe.NullRef`. A user can throw themselves if needed/appropriate.

Reply:
Ok, cool, I must have misremembered the argument against then.

Reply:
My main concern was ensuring the correct functionality was exposed in a way that has the least overhead possible.

Reply:
Haven't seen the api review yet... but I guess the advantage over not checking the return from `GetValueRefOrNullRef` and throwing a nullref is it gives the value of the key in the exception, otherwise its easy to spot what nullref'd as its in the name 😅

Though I'd assume you'd probably go for `GetValueRefOrNullRef` and check the return; avoiding all exceptions where possible.

Reply:
Will remove it for now then